adr/ADR-49.md

```diff
@@ -199,57 +200,62 @@ This seems like a reasonable surface area to add, but, it does stretch the behav
 // Client should validate whether AllowMsgCounter and AllowDirect options are enabled.
 func NewCounterFromStream(stream jetstream.Stream) (Counter, error)
 
 // Entry represents a counter value with source tracking
 type Entry struct {
-	Subject string                         // The counter subject
-	Value   *big.Int                       // Current counter value  
-	Sources map[string]map[string]*big.Int // Source contributions (for aggregated counters)
-	Incr    *big.Int                       // Most recent increment value for this entry. Useful for recounting and auditing purposes.
+    Subject string                         // The counter subject
+    Value   *big.Int                       // Current counter value  
+    Sources map[string]map[string]*big.Int // Source contributions (for aggregated counters)
+    Incr    *big.Int                       // Most recent increment value for this entry. Useful for recounting and auditing purposes.
 }
 
 // Counter provides operations for distributed counters
 type Counter interface {
-  // Add increments the counter for the given subject and returns the new total value.
-	Add(ctx context.Context, subject string, value *big.Int) (*big.Int, error)
-
-	// AddInt increments the counter for the given subject and returns the new total value.
-	AddInt(ctx context.Context, subject string, value int) (*big.Int, error)
-
-	// Load returns the current value of the counter for the given subject.
-	Load(ctx context.Context, subject string) (*Value, error)
-
-	// LoadMultiple returns an iterator over counter values for multiple subjects.
-	// Wildcards are supported.
-	LoadMultiple(ctx context.Context, subjects []string) iter.Seq2[*Value, error]
-
-	// GetEntry returns the full entry with value and source history for the given subject.
-	GetEntry(ctx context.Context, subject string) (*Entry, error)
-
-	// GetEntries returns an iterator over counter entries matching the pattern.
-	// Wildcards are supported.
-	GetEntries(ctx context.Context, subjects []string) iter.Seq2[*Entry, error]
+    // Add increments the counter for the given subject and returns the new total value.
+    Add(ctx context.Context, subject string, value *big.Int) (*big.Int, error)
+
+    // AddInt increments the counter for the given subject and returns the new total value.
+	// - Language Specific Variations are acceptable to handle all native integer type numbers.
+    AddInt(ctx context.Context, subject string, value int) (*big.Int, error)
+
+    // Get returns the current big int of the counter for the given subject.
+    // - Use the "no_hdr": true option on the direct get.
+    Get(ctx context.Context, subject string) (*big.Int, error)
+   
+    // GetMultiple returns an iterator over counter big ints for multiple subjects.
+    // - Wildcards are allowed in subjects.
+    // - Use the "no_hdr": true option on the direct get.
+    GetMultiple(ctx context.Context, subjects []string) iter.Seq2[*big.Int, error]
```

Comment:
I really don't think this is enough. If you load multiple values, you have to be able to match a value with a subject.

Reply:
It should probably be a wildcard here not a list of subjects, the only real use is when you say get `counter.>` so you can easily sum up all the regions under `counter` to get a total - so subject won't matter then.

Reply:
Are we sure this is the only use case? Even if it's only a single subject with wildcard, isn't is possible that the user may want to get both aggregated and per-country result with single API call? No subject only gives the aggregate.

Reply:
for the use case where you just get the bigints, I think so? For the case of getting a list of Entries - sure there you can ask subject array.

But for the case where you do a multi get with no_hdr, you can't get the subjects you only get the big ints, that's the use case